src/SimpleAzure.Storage.HybridQueue/HybridQueue.cs

```diff
@@ -131,51 +111,42 @@ public async Task DeleteMessageAsync<T>(HybridMessage<T> hybridMessage, Cancella
         _logger.LogDebug("Deleting a message.");
 
         // We start with any blobs.
-        if (hybridMessage.BlobId.HasValue)
+        if (hybridMessage.BlobId is { } blobId)
         {
             _logger.LogDebug("Deleting message from Blob Container.");
 
-            var blobClient = _blobContainerClient.GetBlobClient(hybridMessage.BlobId.Value.ToString());
-            var blobResponse = await blobClient.DeleteIfExistsAsync(cancellationToken: cancellationToken);
-
+            var blobClient = _blobContainerClient.GetBlobClient(blobId.ToString());
+            var blobResponse = await blobClient.DeleteIfExistsAsync(cancellationToken: cancellationToken).ConfigureAwait(false);
             if (!blobResponse.Value)
             {
                 _logger.LogWarning("Failed to delete message from Blob Container.");
             }
         }
 
-        var queueResponse = await _queueClient.DeleteMessageAsync(hybridMessage.MessageId, hybridMessage.PopeReceipt, cancellationToken);
+        var queueResponse = await _queueClient.DeleteMessageAsync(hybridMessage.MessageId, hybridMessage.PopeReceipt, cancellationToken).ConfigureAwait(false);
 
         _logger.LogDebug("Deleted a message from the queue.");
     }
 
-    /// <inheritdoc />
-    public async Task<HybridMessage<T?>> GetMessageAsync<T>(TimeSpan? visibilityTimeout, CancellationToken cancellationToken)
+    public async Task<HybridMessage<T>?> GetMessageAsync<T>(TimeSpan? visibilityTimeout, CancellationToken cancellationToken)
     {
-        var messages = await GetMessagesAsync<T>(1, visibilityTimeout, cancellationToken);
+        var messages = await GetMessagesAsync<T>(1, visibilityTimeout, cancellationToken).ConfigureAwait(false);
 
-        if (messages?.Any() ?? false)
+        return messages switch
         {
-            if (messages.Count() > 1)
-            {
-                throw new InvalidOperationException($"Expected 1 message but received {messages.Count()} messages");
-            }
-
-            return messages.First();
-        }
-
-        return default;
+            [] => null,
+            [{ } first] => first,
+            _ => throw new InvalidOperationException($"Expected 1 message but received {messages.Length} messages")
+        };
     }
 
-    /// <inheritdoc />
-    public async Task<IEnumerable<HybridMessage<T>>> GetMessagesAsync<T>(
+    public async Task<HybridMessage<T>[]> GetMessagesAsync<T>(

```

Comment:
Ok - here's another thing I totally didn't understand with your commit message -> the task are called 4 times if this was IEnumerable? versus an array?

Reply:
The old code used IEnumerable, which means it had to call .Any, .Count (twice) and .First. All of these Linq methods start enumerating the collection, which could cause who-knows-what to happen. Could start downloading a blob from azure, or some records from a database. Unless absolutely unavoidable, Linq methods like that should not be called (IMO). In this case it was easily avoidable because we already had an array we can use.

Reply:
> which means it had to call .Any, .Count (twice) and .First

I don't understand what you mean, here.

Where was this getting called?

or are you saying: Never return `IEnumerable` for a library method because the consumer could call `.Any()`, `.Count()` or `First()` or a number of other linq methods, forcing the enumeration to kick in?

Reply:
AH! found it:

![image](https://github.com/PureKrome/WorldDomination.SimpleAzure.Storage.HybridQueues/assets/899878/74f74868-5fb7-4cf1-be98-e7d2ad5f5bed)

I thought Any would do 1 enumerator step and then stop.
Count would enumerate through the entire thing -> so that's legit
And First i thought just enumerates, once.

Instead of an Array, what about an `IList` or `ICollection` ?

Reply:
IReadOnlyList is probably a good compromise here. I'm personally still allergic to IEnumerable but I will grant you the perf concerns I have working in VS are not necessarily in proportion to everyone else :)

Reply:
I'll Approve for the time being.

I _think_ I might revert to IROL .. i just have this thing against arrays, directly :(

Reply:
I thought about this more, and IReadOnlyList is probably the right thing. Returning arrays is a little stinky. IEnumerable is just worse :P